pulp_container/app/tasks/sync_stages.py

```diff
@@ -171,6 +171,12 @@ async def run(self):
                         list_dc.extra_data["listed_manifests"].append(listed_manifest)
 
                     else:
+                        # Manifest indices can be signed too. It is not mandatory.
+                        # If signature is available mirror it.
+                        if signature_source is not None:
+                            list_sig_dcs = await self.create_signatures(list_dc, signature_source)
```

Comment:
The `create_signatures` method will need to be adjusted for manifest lists (OCI indexes). It tries to use the digest of a manifest when initializing signatures. There is no digest for the manifest lists in the payload: https://github.com/pulp/pulp_container/blob/343859eedfa64a022a7062c39372e414e89f7b6b/pulp_container/app/tasks/sync_stages.py#L527

Reply:
Invalid.

Reply:
```
list_dc.content.digest
Out[8]: 'sha256:8a2bd13f5d1cbc36c2a0a5ee2fdc544b5320c2393768a3e9b96b476c8faeab68'
```

Reply:
we are retrieving the digest not from the payload but from the declarative content of ML we have just created on the line earlier ` list_dc = self.create_tagged_manifest_list`